### PR TITLE
doc: update mac instructions

### LINKED ISSUE #6897: Can't build i586 on Mac following instructions
using 1.11.99 , following the [installation_mac](http://docs.zephyrproject.org/getting_started/installation_mac.html) in doc.

Everything works exactly right until this step:

```bash
 cp ${ZEPHYR_BASE}/scripts/cross_compiler/x86.config .config
export CT_PREFIX=/Volumes/CrossToolNG
ct-ng oldconfig
```

After `ct-ng oldconfig`, the file .config looks like this:
image
however in document, it said the .config should look like:
```
...
CT_LOCAL_TARBALLS_DIR="/Volumes/CrossToolNG/src"
# CT_SAVE_TARBALLS is not set
CT_WORK_DIR="${CT_TOP_DIR}/.build"
CT_PREFIX_DIR="/Volumes/CrossToolNG/x-tools/${CT_TARGET}"
CT_INSTALL_DIR="${CT_PREFIX_DIR}"
# Following options prevent link errors
CT_WANTS_STATIC_LINK=n
CT_CC_STATIC_LIBSTDCXX=n
...
```

I think it might cause by the document is behind latest source code, so I run `ct-ng build`.
When finished, the toolchain should be available under /Volumes/CrossToolNG/x-tools, but in my /Volumes/CrossToolNG there is just a i586-zephyr-elf folder.
```
tree -L 2 -a
.
├── .DS_Store
├── .Trashes [error opening dir]
├── .fseventsd
│   ├── 00000000008588fa
....
│   └── fseventsd-uuid
├── build
│   ├── .build
│   ├── .config
│   ├── .config.old
│   ├── build.log.bak
│   ├── build.log_04031008
│   ├── config -> /usr/local/lib/crosstool-ng-1.23.0/config
│   └── config.gen
└── i586-zephyr-elf
    ├── bin
    ├── build.log.bz2
    ├── i586-zephyr-elf
    ├── include
    ├── lib
    ├── libexec
    └── share
```
There is no such x-tools.
I think it might build successfully. but when I try to run samples/hello_world, 
```
export ZEPHYR_TOOLCHAIN_VARIANT=xtools
export XTOOLS_TOOLCHAIN_PATH=/Volumes/CrossToolNG
export ZEPHYR_TOOLCHAIN_VARIANT=xtools
export ZEPHYR_SDK_INSTALL_DIR=/Volumes/CrossToolNG
cd $ZEPHYR_BASE/samples/hello_world
mkdir build && cd build
```

Then I run ` cmake -GNinja -DBOARD=qemu_x86 .. ` and get this:

```
CMake Deprecation Warning at /Users/***/project/zephyr/cmake/app/boilerplate.cmake:38 (cmake_policy):
  The OLD behavior for policy CMP0000 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  CMakeLists.txt:2 (include)


-- Found PythonInterp: /usr/local/bin/python3.4 (found suitable version "3.4.2", minimum required is "3.4")
-- Selected BOARD qemu_x86
Zephyr version: 1.11.99
CMake Error at /Users/***/project/zephyr/cmake/host-tools-zephyr.cmake:24 (message):
  The file '/Volumes/CrossToolNG/sdk_version' was not found.  Is
  ZEPHYR_SDK_INSTALL_DIR=/Volumes/CrossToolNG misconfigured?
Call Stack (most recent call first):
  /Users/***/project/zephyr/cmake/host-tools.cmake:1 (include)
  /Users/***/project/zephyr/cmake/app/boilerplate.cmake:240 (include)
  CMakeLists.txt:2 (include)


-- Configuring incomplete, errors occurred!
```

### LINKED ISSUE #6912: OS X: setup instructions are wrong
`brew install cmake ninja dfu-util doxygen qemu dtc python3 gperf
curl -O 'https://bootstrap.pypa.io/get-pip.py'
./get-pip.py
rm get-pip.py
cd ~/zephyr   # or to the folder where you cloned the zephyr repo
pip3 install --user -r scripts/requirements.txt`

The ./get-pip.py line does not work, the script is not executable and requires sudo anyway.

### PR DESCRIPTION
**Pull request by nashif**
Use sdn-ng scripts to build toolchain instead of repeating them here.

Fixes #6912
Fixes #6897

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
Looks good to me. I followed the links and took a look at them, seems straight forward enough. That said I haven't tried building the toolchains myself.

File: doc/getting_started/installation_mac.rst [hunk 1]
```diff
@@ -108,114 +108,21 @@ To build for the ARM-based Nordic nRF52 Development Kit:
 Setting Up the Toolchain
 ************************
 
-Install tools needed for building the toolchain (if needed):
+In case a toolchain is not available for the board you are using, you can build
+a the toolchain from scratch using crosstool-NG. Follow the steps on the
```
Comment (dbkinder) [inline doc/getting_started/installation_mac.rst]: change ``a the toolchain`` to ``a toolchain``

File: doc/getting_started/installation_mac.rst [hunk 2]
```diff
@@ -108,114 +108,21 @@ To build for the ARM-based Nordic nRF52 Development Kit:
...
+crosstool-NG website to prepare your `host
```
Comment (dbkinder) [inline doc/getting_started/installation_mac.rst]: as written, only the word ``host`` would be clickable... better would be include ``prepare your`` too:

    website to `prepare your host
    <http://crosstool-ng.github.io/docs/os-setup/>`_

File: doc/getting_started/installation_mac.rst [hunk 3]
```diff
@@ -108,114 +108,21 @@ To build for the ARM-based Nordic nRF52 Development Kit:
...
-=============================
-
-In the Zephyr kernel source tree we provide configurations for NIOS-II and
-X86 that can be used to preselect the options needed for building the toolchain.
-
-The configuration files can be found in
-:file:`${ZEPHYR_BASE}/scripts/cross_compiler/`.
-
-Currently the following configurations are provided:
+Follow the steps `here
```
Comment (dbkinder) [inline doc/getting_started/installation_mac.rst]: Better would be:

    Follow the `Zephyr SDK with Crosstool NG instructions
    <https://github.com/zephyrproject-rtos/sdk-ng/blob/master/README.md>`_ to build ...

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by dbkinder — APPROVED** [review]
thanks!

**Pull request merged**